sdc/hiframes/pd_dataframe_ext.py

```diff
@@ -76,15 +101,23 @@ def codegen(context, builder, signature, args):
         dataframe = cgutils.create_struct_proxy(
             signature.return_type)(context, builder)
 
+        data_list_type = [types.List(typ) for typ in data_typs_map.keys()]
```

Comment:
Are you sure that the order of types in the dict will always match the order of columns in DF?

Reply:
Maybe more reliable to create extra list of types as it was done in the model. Then use the list for ordered iterating.